src/test/java/io/mathdojo/QuestionFunctionTest.java

```diff
@@ -1,39 +1,130 @@
 package io.mathdojo;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+import java.net.URISyntaxException;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Logger;
 
+import org.junit.Before;
 import org.junit.Ignore;
 import org.junit.Test;
 import org.springframework.cloud.function.adapter.azure.AzureSpringBootRequestHandler;
 
+import com.microsoft.azure.functions.ExecutionContext;
+
 public class QuestionFunctionTest {
 
-	@Test
-	public void test() {
-		Question question = mock(Question.class);
-		when(question.getQuestionTitle()).thenReturn("test");
-		QuestionFunction qf = new QuestionFunction();
-		qf.repository = mock(MathDojoQuestionRepository.class);
+	private Question question;
+	private QuestionFunction qf = new QuestionFunction();
+	private MathDojoQuestionRepository repo = mock(MathDojoQuestionRepository.class);
+	private MathDojoTopicRepository trepo = mock(MathDojoTopicRepository.class);
+	private Topic topic = new Topic();
+	Map<String, String> headers;
+	Map<String, String> queryParams;
+
+	@Before
+	public void setUp() {
+		question = new Question();
+		question.setQuestionTitle("test");
+		question.setDifficulty("easy");
+		question.setId("test");
+		qf.repository = repo;
+		qf.tRepository = trepo;
 		List<Question> questionList = new ArrayList<Question>();
 		questionList.add(question);
-		when(qf.repository.findAll()).thenReturn(questionList);
-		Question result = qf.getQuestion().apply(new GetObject());
-		assertEquals(result.getQuestionTitle(), "test");
+		when(repo.findByQuestionTitle("test")).thenReturn(questionList);
+		when(repo.findById("test")).thenReturn(Optional.of(question));
+		when(repo.findAllById(new ArrayList<String>(Arrays.asList("test")))).thenReturn(questionList);
+		topic.setId("test");
+		topic.setName("test");
+		topic.setTitle("test");
+		topic.setQuestions(new ArrayList<String>(Arrays.asList("test")));
+		List<Topic> topicList = new ArrayList<Topic>();
+		topicList.add(topic);
+		when(trepo.findByTopicTitle("test")).thenReturn(topicList);
+		when(trepo.findById("test")).thenReturn(Optional.of(topic));
+		headers = new HashMap<>();
+		queryParams = new HashMap<>();
+
 	}
 
 	@Test
+	public void testQuestionAndTopicFunctions() {
```

Comment:
Really great call writing unit tests but for readability reasons these should be in separate functions. That way, if something fails in the future, it'll be very to see where we need to focus our debugging efforts. As it is now, you'd have to sift through all these assertions to understand if any preconditions caused them to fail.

Reply:
will do